docs/testplan/platform/pmon-test-plan.md

```diff
@@ -0,0 +1,227 @@
+- [Overview](#overview)
+  - [Scope](#scope)
+  - [Testbed](#testbed)
+- [Test setup](#test-setup)
+- [Test cases](#test-cases)
+  - [Test case # 1 All expected daemons are running](#test-case-#-1-all-expected-daemons-are-running)
+  - [Test case # 2 Data for chassisd daemon](#test-case-#-2-data-for-chassisd-daemon)
+  - [Test case # 3 Data for lm-sensors daemon](#test-case-#-3-data-for-lm-sensors-daemon)
+  - [Test case # 4 Data for fancontrol daemon](#test-case-#-4-data-for-fancontrol-daemon)
+  - [Test case # 5 Data for ledd daemon](#test-case-#-5-data-for-ledd-daemon)
+  - [Test case # 6 Data for pcied daemon](#test-case-#-6-data-for-pcied-daemon)
+  - [Test case # 7 Data for psud daemon](#test-case-#-7-data-for-psud-daemon)
+  - [Test case # 8 Data for syseepromd daemon](#test-case-#-8-data-for-syseepromd-daemon)
+  - [Test case # 9 Data for thermalctld daemon](#test-case-#-9-data-for-thermalctld-daemon)
+  - [Test case # 10 Data for xcvrd daemon](#test-case-#-10-data-for-xcvrd-daemon)
+- [Platform daemon test report](#platform-daemon-test-report)
+  - [Platform daemon test result table](#platform-daemon-test-result-table)
+
+# Overview
+The purpose is to test the status of daemons in PMON docker on the SONIC switch DUT.
+
+## Scope
+The test is targeting all expected daemons running inside SONIC PMON docker with fully functioning configuration.  
+The purpose of the test is not to test specific api but to verify the platform daemon status and functionality on each platform.
+1. check if pmon docker is running: this can be covered by the critical process check in sanity_check plugin
+2. check if all expected daemons are running on the DUT based on the SONiC pmon docker configuration file and platform specific configuration files:  
```

Comment:
could you check whether the current **get_pmon_daemon_states**  can cover all your requirement for pmon daemon check?  https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices.py#L571